lnwallet/chancloser/rbf_coop_transitions.go

```diff
@@ -31,20 +33,59 @@ var (
 // sendShutdownEvents is a helper function that returns a set of daemon events
 // we need to emit when we decide that we should send a shutdown message. We'll
 // also mark the channel as borked as well, as at this point, we no longer want
-// to continue with normal operation.
+// to continue with normal operation. This function also returns the actual closee
+// nonce used (either provided or auto-generated) for taproot channels.
 func sendShutdownEvents(chanID lnwire.ChannelID, chanPoint wire.OutPoint,
 	deliveryAddr lnwire.DeliveryAddress, peerPub btcec.PublicKey,
-	postSendEvent fn.Option[ProtocolEvent],
-	chanState ChanStateObserver) (protofsm.DaemonEventSet, error) {
+	postSendEvent fn.Option[ProtocolEvent], chanState ChanStateObserver,
+	env *Environment, localCloseeNonce fn.Option[lnwire.Musig2Nonce],
+) (protofsm.DaemonEventSet, fn.Option[lnwire.Musig2Nonce], error) {
+
+	// Create the shutdown message.
+	shutdownMsg := &lnwire.Shutdown{
+		ChannelID: chanID,
+		Address:   deliveryAddr,
+	}
+
+	none := fn.None[lnwire.Musig2Nonce]()
+
+	// For taproot channels using modern RBF flow, auto-generate closee
+	// nonce if not provided. The shutdown message only contains our closee
+	// nonce - the nonce the remote party will use when they act as closer.
+	if env.IsTaproot() {
+		// If closee nonce not provided, generate one now. Note how we
+		// generate it usingt he RemoteMusigSession, as that'll set our
+		// localNonce, we'll receive their remoteNonce for this session
+		// once we get their ClosingComplete message.
+		if localCloseeNonce.IsNone() {
+			remoteMusig := env.RemoteMusigSession
+			if remoteMusig != nil {
```

Comment:
Why would the `remoteMusig` be `nil` at this stage?

Reply:
Correct that it shouldn't be, this is just a defensive check.